src/account/operations/transfer/high_level/sweep_outputs.rs

```diff
@@ -0,0 +1,412 @@
+
+                output_ids.push(output_id);
+                outputs.push(output);
+
+                if output_ids.len() == (OUTPUT_COUNT_MAX - 1) as usize {
+                    let transaction_id = self
+                        .send_sweep_transaction(address.clone(), output_ids.drain(..), outputs.drain(..))
+                        .await?;
+                    transaction_ids.push(transaction_id);
+                }
+            }
+
+            if !output_ids.is_empty() {
+                let transaction_id = self
+                    .send_sweep_transaction(address.clone(), output_ids.drain(..), outputs.drain(..))
+                    .await?;
+                transaction_ids.push(transaction_id);
+            }
+
+            //  Fetch and burn all foundries we can find
+            if let Address::Alias(alias_address) = &address.inner {
+                let _ = self
+                    .destroy_foundries(remainder_address.bech32_hrp(), alias_address)
+                    .await?;
+            }
+
+            Ok(transaction_ids)
+        }
+        .boxed()
+    }
+
+    pub(crate) async fn send_sweep_transaction(
+        &self,
+        address: AddressWrapper,
+        output_ids: impl IntoIterator<Item = OutputId>,
+        outputs: impl IntoIterator<Item = Output>,
+    ) -> crate::Result<TransactionId> {
+        let mut custom_inputs = Vec::new();
+        let mut custom_outputs = Vec::new();
+
+        match address.inner {
+            Address::Alias(alias_address) => {
+                let (output_id, alias_output) = self
+                    .output_id_and_next_alias_output_state(*alias_address.alias_id())
+                    .await?;
+                custom_inputs.push(output_id);
+                custom_outputs.push(Output::Alias(alias_output));
+            }
+            Address::Nft(nft_address) => {
+                let (output_id, nft_output) = self.output_id_and_nft_output(*nft_address.nft_id()).await?;
+                custom_inputs.push(output_id);
+                custom_outputs.push(Output::Nft(nft_output));
+            }
+            Address::Ed25519(_) => {
+                return Err(Error::BurningFailed(
+                    "Ed25519 address is not intended to be swept".to_string(),
+                ));
+            }
+        }
+
+        custom_inputs.append(&mut output_ids.into_iter().collect::<Vec<_>>());
+        custom_outputs.append(&mut outputs.into_iter().collect::<Vec<_>>());
+
+        let transfer_options = Some(TransferOptions {
+            custom_inputs: Some(custom_inputs),
+            try_fetch_unfound_outputs: true,
+            ..Default::default()
+        });
+
+        let transfer_result = self.send(custom_outputs, transfer_options).await?;
+        match &transfer_result.message_id {
+            Some(message_id) => {
+                let _ = self.client.retry_until_included(message_id, None, None).await?;
+                let _ = self.sync(None).await?;
```

Comment:
Maybe we should add such a method with the transaction id? Shouldn't be done in this PR, but if you think it would be better let's open an issue

Reply:
I think this should be done in a separate PR